WalletWasabi.Gui/Controls/WalletExplorer/WalletViewModel.cs

```diff
@@ -54,23 +56,23 @@ public WalletViewModel(WalletService walletService, bool receiveDominant)
 			// If hardware wallet then we need the Send tab.
 			if (WalletService?.KeyManager?.IsHardwareWallet is true)
 			{
-				sendTab = new SendTabViewModel(this);
+				sendTab = new SendTabViewModel(WalletService);
 				Actions.Add(sendTab);
 			}
 			// If not hardware wallet, but neither watch only then we also need the send tab.
 			else if (WalletService?.KeyManager?.IsWatchOnly is false)
 			{
-				sendTab = new SendTabViewModel(this);
+				sendTab = new SendTabViewModel(WalletService);
 				Actions.Add(sendTab);
 			}
 
-			var receiveTab = new ReceiveTabViewModel(this);
-			var coinjoinTab = new CoinJoinTabViewModel(this);
-			var historyTab = new HistoryTabViewModel(this);
+			var receiveTab = new ReceiveTabViewModel(walletService);
+			var coinjoinTab = new CoinJoinTabViewModel(walletService);
+			var historyTab = new HistoryTabViewModel(walletService);
 
-			var advancedAction = new WalletAdvancedViewModel(this);
-			var infoTab = new WalletInfoViewModel(this);
-			var buildTab = new BuildTabViewModel(this);
+			var advancedAction = new WalletAdvancedViewModel(Name);
+			var infoTab = new WalletInfoViewModel(walletService, Name);
+			var buildTab = new BuildTabViewModel(walletService, Id);
```

Comment:
`BuildTabViewModel` does not contain a constructor that takes 2 arguments.
```suggestion
			var buildTab = new BuildTabViewModel(walletService);
```

Reply:
done